gensim/similarities/docsim.py

```diff
@@ -392,12 +394,18 @@ def __iter__(self):
         # turn off query normalization (vectors in the index are already normalized, save some CPU)
         norm, self.norm = self.norm, False
 
-        for chunk in self.iter_chunks():
-            if chunk.shape[0] > 1:
-                for sim in self[chunk]:
+        with concurrent.futures.ProcessPoolExecutor() as executor:
```

Comment:
Depends on the size of machine, for me it took down a task that took 10hrs down to ~2hrs. 
What is the standard way to measure perf?